### PR TITLE
Switch tool buttons escene view

### PR DESCRIPTION
**Pull request by alonss0**
Left to check why the InvokeSelectionChangedEvent when CurrentMode Set unselects the entity.

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m41y5d Switch tool buttons in scene view](https://app.clickup.com/t/861m41y5d)

**Comment by alonss0** [comment]
In this commit c772a0fcd2f2c5e9ac02cf6355660247a507ea5e I have a blocker, not sure why the entity when click on scene button sets to null, so the method `_editorEvents.InvokeSelectionChangedEvent();` called in line 38 from `GizmoState` is hiding the gizmo, but the current mode of the gizmo is changing as spected.

**Comment by alonss0** [comment]
Hi @cblech, in this commit 1a4bbaa9ad563bd57fcca58e7ef416ee97d8d0ad I have made some changes to check when the pointer in over the buttons in the scene, I have a question: The script `MouseHoverSceneButton` is being added as a component to each button game object on the unity editor, now I need to access its `IsMouseOverGizmoModeMenu` property from the `GeneralInputInteraction`, is there a way to access the property of each one of those buttons script component?

**Comment by cblech** [comment]
I'd recommend adding a list of `MouseHoverSceneButtons` into the `UnityState`. You then need to reference all Scene buttons in there. Then you can access the list via the `unityState` in `GeneralInputInteraction`

File: Assets/Scenes/editor.unity [hunk 1]
```diff
@@ -123,139 +123,1180 @@ NavMeshSettings:
     debug:
       m_Flags: 0
   m_NavMeshData: {fileID: 0}
+--- !u!1 &18110688
```
Comment (cblech) [inline Assets/Scenes/editor.unity]: Please apply the `GizmoModeMenu` to the `SceneTab` prefab

File: Assets/Scripts/EditorState/GizmoState.cs [hunk 2]
```diff
@@ -8,6 +9,8 @@ public class GizmoState
         // Dependencies
         private EditorEvents _editorEvents;
 
+        public static UnityEvent onUpdate = new UnityEvent();
```
Comment (cblech) [inline Assets/Scripts/EditorState/GizmoState.cs]: Please move this into the editor events

**Review by cblech — CHANGES_REQUESTED** [review]

**Review by cblech — APPROVED** [review]

**Pull request merged**